TechAndCheck/zenodotus
# Consolidate author controllers

This PR consolidates all the various author-type controllers into a single one. This reduces a ton of controller and view redundancy, which was having a real effect as some gnarly bugs were hiding in the discrepancies between templates. (And once I consolidate search results in #473, even more redundancy will be removed.)

It consists of just one commit, but that was pretty necessary since so much was interlinked. Much of the diff is just renaming/moving files, deleting files, and updating routes.

You'll notice that the author URLs are now `/authors/:platform/:id` (e.g., `/authors/twitter/abc123`) rather than something like `/twitter_user/:id` or `/youtube_channel/:id`.

**Testing:**
1. `rails t`
2. `./bin/dev`
3. View the MediaVault Dashboard
4. Drill into users, preferably _every_ user type (twitter, facebook, etc.)
5. Search for results across all user and post types (note that the styling is still old, but nothing should break, except a search for `biden` may trigger the problem raised in #464)

Closes #477
